src/sparse/KokkosSparse_BsrMatrix.hpp

```diff
@@ -913,6 +941,36 @@ class BsrMatrix {
 //----------------------------------------------------------------------------
 //----------------------------------------------------------------------------
 
+template <typename ScalarType, typename OrdinalType, class Device,
+          class MemoryTraits, typename SizeType>
+void BsrMatrix<ScalarType, OrdinalType, Device, MemoryTraits,
+               SizeType>::ctor_impl(const std::string& /*label*/,
+                                    OrdinalType nrows,
+                                    OrdinalType ncols,
+                                    size_type annz, ScalarType* val,
+                                    OrdinalType* rows, OrdinalType* cols,
+                                    OrdinalType blockDimIn) {
+  numCols_  = ncols;
+  blockDim_ = blockDimIn;
+
+  // Wrap the raw pointers in unmanaged host Views
+  typename values_type::HostMirror unman_val(val, annz);
+  typename row_map_type::HostMirror unman_rows(rows, nrows + 1);
+  typename index_type::HostMirror unman_cols(cols, ncols);
```

Comment:
Should this be `annz` instead of `ncols`?

Reply:
You are correct that `ncols` is not right  (actually the same issue appears in `Kokkos_BlockCrsMatrix.hpp` ).
It may not be `annz` as the graph is only as the block level.
I will fix that.

Reply:
I updated the constructor (previously calling the function 'ctor_impl') to use the arrays providing entries by coordinates.